cmd/influx/user_resource_mapping.go

```diff
@@ -3,10 +3,74 @@ package main
 import (
 	"context"
 	"fmt"
+	"os"
 
 	"github.com/influxdata/influxdb"
+	"github.com/influxdata/influxdb/cmd/influx/internal"
 )
 
+func membersListF(filter influxdb.UserResourceMappingFilter, flgs Flags) error {
+	ctx := context.Background()
+	mappingSvc, err := newUserResourceMappingService(flgs)
+	if err != nil {
+		return fmt.Errorf("failed to initialize members service client: %v", err)
+	}
+	mps, _, err := mappingSvc.FindUserResourceMappings(ctx, filter)
+	if err != nil {
+		return fmt.Errorf("failed to find members: %v", err)
+	}
+	userSVC, err := newUserService(flgs)
+	if err != nil {
+		return fmt.Errorf("failed to initialize users service client: %v", err)
+	}
+	urs := make([]*influxdb.User, len(mps))
+	ursC := make(chan struct {
+		User  *influxdb.User
+		Index int
+	}, len(mps))
+	errC := make(chan error)
+	for k, v := range mps {
+		go func(k int, v *influxdb.UserResourceMapping) {
+			usr, err := userSVC.FindUserByID(ctx, v.UserID)
+			if err != nil {
+				errC <- fmt.Errorf("failed to retrieve user details: %v", err)
+				return
+			}
+			ursC <- struct {
+				User  *influxdb.User
+				Index int
+			}{
+				User:  usr,
+				Index: k,
+			}
+		}(k, v)
+	}
+	for i := 0; i < len(mps); i++ {
+		select {
+		case err := <-errC:
+			return err
+		case item := <-ursC:
+			urs[item.Index] = item.User
```

Comment:
can we add `<-ctx.Done` as well? if there is ever a panic in the `usersSVC.FindUserByID`, then this will block forever potentially. Not sure if that is a ticker, maybe give it 10 seconds to complete or something 🤷‍♂

Reply:
yeah, that's a good idea

Reply:
added another ticket to wrap up the timeout: https://github.com/influxdata/influxdb/issues/15375